core/table.ts

```diff
@@ -492,3 +554,38 @@ export class TableContext implements ITableContext {
     return "";
   }
 }
+
+function indexAssertionSql(ctx: AssertionContext, table: dataform.ITarget, indexCols: string[]) {
```

Comment:
don't love that this is in this file, feels like it should be part of adapter (would be nice to constrain all SQL generating code to a single place!). However, all that code is a mess so not going to block